Adding DPUB use cases (#1760)
act-rules/act-rules.github.io: _rules/link-non-empty-accessible-name-c487ae.md

## Applicability

> This rule applies to any [HTML element][] that is an [inheriting semantic][inheriting semantic] `link` [included in the accessibility tree][].

Reviewer: Think you can simplify this:

```suggestion
This rule applies to any [HTML element][] that is an [inheriting semantic][] `link` [included in the accessibility tree][].
```

> This rule applies to any [HTML element][] that is an [inheriting semantic][inheriting semantic role] `link` [included in the accessibility tree][].

Reviewer: Suggest you do them the other way around:

```suggestion
This rule applies to any [HTML element][] that is an [inheriting semantic][] `link` [included in the accessibility tree][].
```

- There are assistive technologies that do not support using the `title` attribute for an [accessible name][], or in which this feature can be disabled.
- For `area` elements that have an `href` attribute, but are not nested inside a `map` element, there are differences between browsers and assistive technology on if the `area` is [included in the accessibility tree][].
- Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some [semantic][semantic role] `link` elements can fail this rule with some technology but users of other technologies would not experience any accessibility issue.

> - Accessibility support for [inheriting semantic][] links may vary depending on the assistive technology in use.

Reviewer 2: ```suggestion
- Accessibility support for some elements inheriting the semantic role of `link` (e.g. elements with `doc-*` attributes) may vary depending on the assistive technology in use.
```

> - Accessibility support for [inheriting semantic][] links may vary depending on the assistive technology in use.

Author: ```suggestion
- Accessibility support for some elements inheriting the semantic role of `link` (e.g. elements with `doc-*` attributes) may vary depending on the assistive technology in use.
```
I seem to have dropped this accidentally @carlosapaduarte  Bringing it back.

> - Accessibility support for [inheriting semantic][inheriting-semantic] links may vary depending on the assistive technology in use.

Reviewer: ```suggestion
- Accessibility support for [inheriting semantic][] links may vary depending on the assistive technology in use.
```

</map>
```

#### Passed Example 11

This `span` element has an [inheriting semantic role][] of `link` and an [accessible name] given by its child text node.

```html

> See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref">ACT rules</span>]

Reviewer: I'm a little hesitant to say this passes SC 4.1.2. This wouldn't be accessibility supported. Perhaps what we need to do is put these on actual anchor elements?

> See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref link">ACT rules</span

Author: ```suggestion
See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref">ACT rules</span>]
```

</map>
```

#### Passed Example 11

> This `a` element has an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute as well as an [accessible name] given by its child text node.

Reviewer 2: To be consistent with the way the applicability is written. 

```suggestion
This `a` element is an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute as well as an [accessible name] given by its child text node.
```

> This `a` element has an [inheriting semantic role][] of `link` given by its `role="doc-biblioref"` attribute as well as an [accessible name] given by its child text node.

Reviewer: ```suggestion
This `a` element has an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute as well as an [accessible name] given by its child text node.
```

```html

> See [<a onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref">ACT rules</a>]

Reviewer 3: This is not a `span` element 🙈 
```suggestion
See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref">ACT rules</span>].
```

<a href="https://www.w3.org/WAI" role="none"> </a>
```

### Failed Example 11

This `span` element has an [inheriting semantic role] of `link` and an empty [accessible name].

```html

> See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref link"></span

Author: ```suggestion
See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref"></span>]
```

<a href="https://www.w3.org/WAI" role="none"> </a>
```

#### Failed Example 11

> This `a` element has an [inheriting semantic role][] of `link` given by its `role="doc-biblioref"` attribute but it has an empty [accessible name].

Reviewer: ```suggestion
This `a` element has an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute but it has an empty [accessible name].
```

```html

> See [<a onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref"></a>]

Reviewer 3: ```suggestion
See [<span onclick="location.href='https://act-rules.github.io/'" role="doc-biblioref"></span>].
```

Reviewer 3: Maybe adding an image inside the `span` here. Visually, nothing is displayed and there is almost nowhere to actually click, which is at least weird.
Adding [this image](https://github.com/act-rules/act-rules.github.io/blob/develop/test-assets/shared/act-logo.png) with an empty `alt` would be better, I think.

<a href="https://www.w3.org/WAI" role="none"> </a>
```

#### Failed Example 11

> This `a` element has an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute but it has an empty [accessible name].

Reviewer 2: To be consistent with the way the applicability is written. 

```suggestion
This `a` element is an [inheriting semantic][] `link` given by its `role="doc-biblioref"` attribute but it has an empty [accessible name].
```

[explicit role]: #explicit-role 'Definition of Explicit Role'
[focusable]: #focusable 'Definition of focusable'
[included in the accessibility tree]: #included-in-the-accessibility-tree 'Definition of included in the accessibility tree'

> [inheriting semantic role]: #inheriting-semantic-role 'Definition of inheriting semantic role'

Reviewer 3: ```suggestion
[inheriting semantic role]: #inheriting-semantic 'Definition of inheriting semantic role'
```
The key is only `inheriting-semantic`, that should fix the test 🤞
